exercise/MyAgenda.res

```diff
@@ -9,10 +9,37 @@ module Event = {
 }
 
 let oneRecommendationEveryDay = (~maxDistanceInKm, ~maxPrice) => {
-  // effacer tout ce qui suit et le remplacer par une implémentation correcte
-  // utiliser la fonction API.Event.nextEvents pour récupérer les événements
-  // puis les autres fonctions du module API pour obtenir les informations manquantes
-  ignore(maxDistanceInKm)
-  ignore(maxPrice)
-  failwith("TODO")
+  // Get the current user's location
+  let currentUserLocation = API.Location.getCurrentLocation()
+  
+  // Retrieve a list of upcoming events
+  let events = API.Event.nextEvents()
+  
+  events
+  // Filter events based on maximum distance and price
+  |> List.filter(event => {
```

Comment:
it's not far, but events is an array, so make use the `Array` module and don't use the deprecated pipe-last (`|>`), either use the `->` pipe-first, or just regular function calls.